packaging/linux/post_install.sh

```diff
@@ -1,56 +1,79 @@
 #! /usr/bin/env bash
 
 # This script handles distro-independent Linux post-install tasks. Currently
-# that means creating the /keybase directory and making it writable. The deb-
+# that means managing the /keybase redirector mountpoint. The deb-
 # and rpm-specific post-install scripts call into this after doing their
 # distro-specific work, which is mainly setting up package repos for updates.
 
 set -u
 
-# Create the /keybase symlink to the first mountpoint, if it doesn't
-# already exist. We can't use a regular [ -e ... ] check, because
-# stat'ing /keybase fails with a permissions error when kbfsfuse is
-# mounted, so that check always returns false. Instead we check
-# whether mkdir succeeds.
-rootlink="/keybase"
-vardir="/var/lib/keybase"
-mount1="$vardir/mount1"
-sample="/opt/keybase/mount-readme"
-khuser="keybasehelper"
-khbin="/usr/bin/keybase-mount-helper"
-
-# Create the keybasehelper system user, without login privileges.
-if useradd --system -s /bin/false -U -M $khuser &> /dev/null ; then
-    echo Created $khuser system user for managing mountpoints.
-fi
+rootmount="/keybase"
+krbin="/usr/bin/keybase-redirector"
 
-chown "$khuser":"$khuser" "$khbin"
-chmod 4755 "$khbin"
-if mkdir $vardir &> /dev/null ; then
-    ln -s "$sample" "$mount1"
-    chown -R "$khuser":"$khuser" "$vardir"
-fi
+vardirDeprecated="/var/lib/keybase"
+khuserDeprecated="keybasehelper"
+khbinDeprecated="/usr/bin/keybase-mount-helper"
+optDeprecated="/opt/keybase/mount-readme"
+
+chown root:root "$krbin"
+chmod 4755 "$krbin"
 
-currlink=`readlink $rootlink`
-if [ -z "$currlink" ] ; then
-    if fusermount -uz "$rootlink" &> /dev/null ; then
+run_redirector() {
+  logdir="${XDG_CACHE_HOME:-$HOME/.cache}/keybase"
+  mkdir -p "$logdir"
+  nohup "$krbin" "$rootmount" >> $logdir/keybase.redirector.log 2>&1 &
+}
+
+currlink=`readlink -m "$rootmount"`
+if [ -n "$currlink" ] ; then
+    # Upgrade from a rootlink-based build.
+    if rm "$rootmount" &> /dev/null ; then
+        echo Replacing old $rootmount symlink.
+    fi
+    mounts=`cat /proc/mounts | awk '{print $2}'`
+    # TODO: Set $IFS in case any of the mountpoints contain a space?
+    # Probably not a big deal since all this loop does is start the
+    # redirector, which would also be done by the next call to
+    # `run_keybase`.
+    for m in $mounts; do
+      if [ "$m" = "$currlink" ]; then
+        echo Starting root redirector at $rootmount.
+        run_redirector
```

Comment:
What is the significance of this loop? It doesn't pass anything into the `run_redirector` function...so is it just a filter?

Reply:
It's just trying to figure out whether the user currently has anything mounted at the ultimate target of the keybase symlink.  Normally I would use `mountpoint` but it doesn't work from root since KBFS doesn't do `allow_root`.  I'll add a comment.